c10/util/Optional.h

```diff
@@ -508,6 +510,22 @@ class optional : private OptionalBase<T> {
         : detail_::convert<T>(constexpr_forward<V>(v));
   }
 
+  template <class F>
+  constexpr T value_or_else(F&& func) const& {
```

Comment:
Isn't this, like, super cutting edge? I didn't see `value_or_else` in C++20, looks like there's some WG paper http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2218r0.pdf submitted literally like a month ago. Don't we have an implicit goal that c10::optional should track the standard std::optional?

Reply:
yes it's relatively new. We can remove it again if the proposal doesn't make it in or if we want to replace c10::optional with std::optional before that. Should be a relatively simple string replacement to remove it if we need to.